ai/integrations/anthropic/tests/test_anthropic_utils.py

```diff
@@ -267,16 +273,32 @@ def test_generate_tool_call_messages_with_invalid_tool_use_block(mock_client, du
         ("CSV", RETRIEVER_OUTPUT_CSV),
     ],
 )
-def test_generate_tool_call_messages_with_tracing(dummy_history, format: str, function_output: str):
+@pytest.mark.parametrize(
+    "data_type,full_data_type,return_params",
+    [
+        (ColumnTypeName.TABLE_TYPE, RETRIEVER_TABLE_FULL_DATA_TYPE, RETRIEVER_TABLE_RETURN_PARAMS),
+    ],
+)
+def test_generate_tool_call_messages_with_tracing(
+    dummy_history, format, function_output, data_type, full_data_type, return_params
+):
     with mock.patch(
         "unitycatalog.ai.core.databricks.get_default_databricks_workspace_client",
         return_value=mock.Mock(),
     ):
+        function_mock = mock.MagicMock()
```

Comment:
Can we use a spec entry here to ensure that we're mocking the correct interface? `spec=FunctionInfo`
specifying `autospec=True` in the Mock definition should help to ensure that we're constructing the mock interface with the same properties as the underlying FunctionInfo classdef as well. 

For instance:

```python
function_mock = Mock(
    spec=FunctionInfo,
    name=f"catalog.schema.retriever_tool_{format}",
    full_name=f"catalog.schema.retriever_tool_{format}",
    data_type=data_type,
    full_data_type=full_data_type,
    return_params=return_params,
    autospec=True
)
```